evalml/tests/component_tests/test_components.py

```diff
@@ -1289,6 +1289,8 @@ def test_estimators_accept_all_kwargs(
         params = estimator._component_obj.get_params()
         if "random_state" in params:
             del params["random_state"]
+        if estimator.name == "XGBoost Classifier":
```

Comment:
nit-pick: Why not pop `use_label_encoder` from the kwargs rather than changing the test?